### PR TITLE
board: cavs15: Add a option to control signing ways

### LINKED ISSUE #31819: intel_adsp_cavs15: signing not correct thus download firmware failed
**Describe the bug**
When run testcases we **use west script(sign.py) to signing** with image (no error and show completed) then invoke download script to load firmware, it has some error log at firmware loaded step. By debug, found signing not correct in fact(even if show completed without error), if we **use command to signing by ourselves without west script(sign.py)** it will download successful.

**To Reproduce**
1.  west build -b intel_adsp_cavs15 tests/kernel/thread/thread_apis/
2. west sign -t rimage -p ../modules/audio/sof/zephyr/ext/rimage/build/rimage -D ../modules/audio/sof/zephyr/ext/rimage/config/ -- -k ../modules/audio/sof/keys/otc_private_key.pem
3. boards/xtensa/intel_adsp_cavs15/tools/fw_loader.py -f /home/ztest/work/zephyrproject/zephyr/build/zephyr/zephyr.ri

1.  west build -b intel_adsp_cavs15 tests/kernel/thread/thread_apis/
2. ../modules/audio/sof/zephyr/ext/rimage/build/rimage -k ../modules/audio/sof/keys/otc_private_key.pem -o /home/ztest/work/zephyrproject/zephyr/build/zephyr/zephyr.ri -c ../modules/audio/sof/zephyr/ext/rimage/config/apl.toml -i 3 -e /home/ztest/work/zephyrproject/zephyr/build/zephyr/bootloader.elf.mod /home/ztest/work/zephyrproject/zephyr/build/zephyr/zephyr.elf.mod
3. boards/xtensa/intel_adsp_cavs15/tools/fw_loader.py -f /home/ztest/work/zephyrproject/zephyr/build/zephyr/zephyr.ri
4. load firmware successful

**Environment (please complete the following information):**
OS: Fedora28
Toolchain: Zephyr-sdk-0.12.1
Commit ID: 3f1d1f5773ddaf

### PR DESCRIPTION
**Pull request by KangJianX**
Fixed [#31819](https://github.com/zephyrproject-rtos/zephyr/issues/31819)

Zephyr testcases(not SOF case) not use kernel DSP driver to load image
on ADSP board, thus do not need signing with xman. So add a 'no-manifest'
to specify signing without xman in image. If use DSP driver load image,
we should not specify.
And fixed a syntax error in script when download firmware. When copy the data to buffer, the old script can not assign sequence of same size, so specify size.

Signed-off-by: Jian Kang <jianx.kang@intel.com>

### CODE REVIEW

File: scripts/west_commands/sign.py [hunk 1]
```diff
@@ -112,6 +112,8 @@ def do_add_parser(self, parser_adder):
                            help='''path to the tool itself, if needed''')
         group.add_argument('-D', '--tool-data', default=None,
                            help='''path to tool data/configuration directory, if needed''')
+        group.add_argument('-X', '--no-xman', action='store_true',
+                           help='''signing image without xman, if needed''')
```
Comment (lyakh) [inline scripts/west_commands/sign.py:116]: maybe just `'''do not add extended manifest'''`

File: boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ def copy(self, data, size):
             raise ValueError("Not enough buffer. allocated: %d requested: %d"
                              % (self.alloc_size, size))
         logging.debug("Copying Data to DMA buffer")
-        self.buf[:] = data[:]
+        self.buf[:size] = data[:size]
```
Comment (lyakh) [inline boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: is this related or needed? Was the original version wrong?

**Review by lyakh — COMMENTED** [review]
looks good! Just a bit of clarification / clean up maybe

File: boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py (hunk 2 above)
Comment (KangJianX) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: The original version wrong, it can only assign sequence of same size, buffer and data not same size, so need this change.

File: scripts/west_commands/sign.py (hunk 1 above)
Comment (KangJianX) [reply scripts/west_commands/sign.py]: Thanks, improve the description.

File: boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py (hunk 2 above)
Comment (lyakh) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: it changed now because you removed the manifest from the image?

Comment (KangJianX) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: This change is not due to remove manifest. Even if add manifest, the buffer and data size still not same, it is fix for load firmware process.

Comment (lyakh) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: ok, so it isn't really related to the main topic of this commit...

Comment (KangJianX) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: yes, should I use a new commit for this change?

Comment (lyakh) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: I think that would be easier to understand, yes. Thanks!

Comment (KangJianX) [reply boards/xtensa/intel_adsp_cavs15/tools/lib/stream_desc.py:41]: done, do you think it's ok?

**Review by lyakh — APPROVED** [review]
Thanks for addressing comments!

**Review by mbolivar-nordic — DISMISSED** [review]
`--no-xman` should not be added to the common arguments, because it doesn't apply to imgtool. Please use `tool_args`.

**Comment by KangJianX** [comment]
@mbolivar-nordic 
> `--no-xman` should not be added to the common arguments, because it doesn't apply to imgtool. Please use `tool_args`.

source code:
```
 sign_base = ([tool_path] + args.tool_args +
                     ['-o', out_bin] +  conf_path_cmd + ['-i', '3', '-e'] +
                     [bootloader, kernel])

```
From source code, the `args.tool_args` are used by imgtool already, but `--no-xman` should not be applied in imgtool, so I added a option that can be use if needed.

**Review by andyross — APPROVED** [review]
This seems to be at best an optimization of just a few bytes in image transferred over DMA.  The manifest doesn't go into SRAM and shouldn't change runtime behavior at all.  Not that it's a bad idea, but why do we care?  Is something breaking? +1 regardless.

That said: Marti's criticism is correct: this is a tool-specific thing with rimage and doesn't belong in the global signing arguments.

**Comment by KangJianX** [comment]
@mbolivar-nordic address your comment, could you help to approve or add any comments?

**Comment by LeiW000** [comment]
I would like to see it fxied in 2.5 release. @mbolivar-nordic , please take look at the code change again.

**Comment by KangJianX** [comment]
@mbolivar-nordic Hi, could you help to review again?

File: boards/xtensa/intel_adsp_cavs15/tools/flash.sh [hunk 3]
```diff
@@ -13,7 +13,7 @@ if [ -z "$2" ]
 elif [ -n "$3" ] && [ -n "$4" ]
   then
     echo "Signing with key " $key
-    west sign -d ${BUILD} -t rimage -p $4 -D $3 -- -k $2
+    west sign -d ${BUILD} -t rimage -p $4 -D $3 -- -k $2 no-manifest
```
Comment (lyakh) [inline boards/xtensa/intel_adsp_cavs15/tools/flash.sh:16]: Shouldn't this be `--no-manifest` as a proper command-line parameter?

**Review by lyakh — CHANGES_REQUESTED** [review]

File: boards/xtensa/intel_adsp_cavs15/tools/flash.sh (hunk 3 above)
Comment (nashif) [reply boards/xtensa/intel_adsp_cavs15/tools/flash.sh:16]: done

**Review by nashif — APPROVED** [review]

**Review by lyakh — APPROVED** [review]
